### ISSUE TITLE
lua-protobuf dependency upgrade

### ISSUE THREAD

**Issue by fatihusta** [yangxikun/opentelemetry-lua#98]
Hi,
Is it possible to upgrade lua-protobuf version? Current version is 0.3.3 last commit is 3 years ago, last released version is 0.5.1.
Why I need this? I'm using another project and the project depends lua-protobuf-0.5.0 also depends opentelemetry-lua.  So [issue](https://github.com/apache/apisix/issues/9683) was came from this point. I fixed with workaround solution. But now I can't use this workaround with new version of apisix.

```
Will not remove installed versions of lua-protobuf.
Removing them would break dependencies for:
lua-resty-etcd 1.10.5-0
opentelemetry-lua 0.2-3

Use --force to force removal (warning: this may break modules).
Failed removing.
```

**Comment by yangxikun** [comment]
I use `apache/apisix:3.9.1-debian` run a demo, to test the opentelemetry plugin, it works fine.

In the docker image, there are three lua-protobuf version installed:

<img width="594" alt="image" src="https://github.com/yangxikun/opentelemetry-lua/assets/3364160/7531571e-4ffb-4921-bf26-09f582567728">

also the three so library:

![image](https://github.com/yangxikun/opentelemetry-lua/assets/3364160/9daf423f-15f8-46df-a528-ade311b1748d)

And the nginx worker use the highest version lua-protobuf library, `su nobody -s /bin/sh -c "pmap -p 49"`:

![image](https://github.com/yangxikun/opentelemetry-lua/assets/3364160/eb46b2b9-43e3-427c-b280-8e8761158221)

**Comment by fatihusta** [comment]
I removed my old solution and I tested again. It looks fixed with lua-protbuf-5.0.  Thank you for your response.
